core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -276,8 +276,47 @@ object BitcoinSignerSingle {
                                   signer,
                                   conditionalPath)
 
+    val txToSign = spendingInfo.output.scriptPubKey match {
```

Comment:
Is there a reason that `P2WPKH` doesn't need to be accounted for here as well?

Reply:
I am not sure, but it works if I make it only `P2WPKH`

Reply:
do you mean only `P2WSH`?

Reply:
maybe this has to do with the fact that what you sign in the `p2wpkh` case is the reconstructed `p2pkh` which doesn't involve witness stuff